apisix/plugin.lua

```diff
@@ -344,6 +346,15 @@ function _M.load(config)
             if not ok then
                 core.log.error("failed to load plugins: ", err)
             end
+
+            local enabled = core.table.array_find(http_plugin_names, "prometheus") ~= nil
+            local active  = exporter.get_prometheus() ~= nil
```

Comment:
I think we don't need to get the active status.
Since the `exporter.destroy` is harmless.

Reply:
you are right

Reply:
no I think the second if check needs the `active` status. Consider this case:

- apisix is running with prometheus active and enabled
- hit plugin reload

In this case `enabled == true` and `active == true`. But we need not call `http_init()` in prometheus because it is already running during this operation.

Reply:
Got it.